lib/ansible/module_utils/csharp/Ansible.Basic.cs

```diff
@@ -315,7 +315,11 @@ public void LogEvent(string message, EventLogEntryType logEntryType = EventLogEn
             using (EventLog eventLog = new EventLog("Application"))
             {
                 eventLog.Source = logSource;
-                eventLog.WriteEntry(message, logEntryType, 0);
+                try
+                {
+                    eventLog.WriteEntry(message, logEntryType, 0);
+                }
+                catch (System.InvalidOperationException) { }  // Ignore permission errors on the Application event log
```

Comment:
Is this always what's thrown for a permissions failure (I would've expected `UnauthorizedAccessException`)? Wondering if we should just always silently fail on any kind of event log write failure (or warn if it's anything other than `InvalidOperationException`) just to cover the bases...

Reply:
I tried to use `System.Security.SecurityException` and `UnauthorizedAccessException` but they aren't caught in this case. The [docs for WriteEntry](https://docs.microsoft.com/en-us/dotnet/api/system.diagnostics.eventlog.writeentry?view=netframework-4.7.2#System_Diagnostics_EventLog_WriteEntry_System_String_System_Diagnostics_EventLogEntryType_System_Int32_) also state `InvalidOperationException` is raised when the event log reg key could not be opened.

I'm happy to add a warning to other exceptions but I'm not comfortable with the idea of ignoring all exceptions.